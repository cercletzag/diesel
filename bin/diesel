#!/bin/sh
cd diesel_cli && cargo run -- $@
