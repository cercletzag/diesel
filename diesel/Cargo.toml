[package]
name = "diesel"
version = "0.4.1"
authors = ["Sean Griffin <sean@seantheprogrammer.com>"]
license = "MIT OR Apache-2.0"
description = "A safe, extensible ORM and Query builder"
documentation = "http://sgrif.github.io/diesel/diesel/index.html"
homepage = "https://github.com/sgrif/diesel"
repository = "https://github.com/sgrif/diesel"
keywords = ["orm", "database", "postgres", "postgresql", "sql"]

[dependencies]
libc = "0.2.*"
pq-sys = { version = "^0.2.0", optional = true }
libsqlite3-sys = { version = "^0.4.0", optional = true }
byteorder = "0.3.*"
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git", optional = true }
chrono = { version = "^0.2.17", optional = true }

[dev-dependencies]
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git" }
dotenv = "^0.6.0"
tempdir = "^0.3.4"

[features]
default = ["postgres"]
unstable = []
large-tables = []
postgres = ["pq-sys"]
sqlite = ["libsqlite3-sys"]
