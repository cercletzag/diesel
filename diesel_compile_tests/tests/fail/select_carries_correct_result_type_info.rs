extern crate diesel;

use diesel::*;

table! {
    users {
        id -> Integer,
        name -> VarChar,
    }
}

fn main() {
    use self::users::dsl::*;

    let connection = PgConnection::establish("").unwrap();
    let select_id = users.select(id);
    let select_name = users.select(name);

    let ids = select_name.load::<i32>(&connection);
    let names = select_id.load::<String>(&connection);
}
