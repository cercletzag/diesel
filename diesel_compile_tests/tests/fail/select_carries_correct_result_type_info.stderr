error[E0277]: the trait bound `i32: diesel::deserialize::FromSql<diesel::sql_types::Text, _>` is not satisfied
  --> $DIR/select_carries_correct_result_type_info.rs:19:27
   |
19 |     let ids = select_name.load::<i32>(&connection);
   |                           ^^^^ the trait `diesel::deserialize::FromSql<diesel::sql_types::Text, _>` is not implemented for `i32`
   |
   = help: the following implementations were found:
             <i32 as diesel::deserialize::FromSql<diesel::sql_types::Integer, DB>>
             <i32 as diesel::deserialize::FromSql<diesel::sql_types::Integer, diesel::mysql::Mysql>>
             <i32 as diesel::deserialize::FromSql<diesel::sql_types::Integer, diesel::sqlite::Sqlite>>
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Text, _>` for `i32`
   = note: required because of the requirements on the impl of `diesel::deserialize::FromSqlRow<diesel::sql_types::Text, _>` for `i32`
   = note: required because of the requirements on the impl of `diesel::query_dsl::LoadQuery<_, i32>` for `diesel::query_builder::SelectStatement<users::table, diesel::query_builder::select_clause::SelectClause<users::columns::name>>`

error[E0277]: the trait bound `*const str: diesel::deserialize::FromSql<diesel::sql_types::Integer, _>` is not satisfied
  --> $DIR/select_carries_correct_result_type_info.rs:20:27
   |
20 |     let names = select_id.load::<String>(&connection);
   |                           ^^^^ the trait `diesel::deserialize::FromSql<diesel::sql_types::Integer, _>` is not implemented for `*const str`
   |
   = help: the following implementations were found:
             <*const [u8] as diesel::deserialize::FromSql<diesel::sql_types::Binary, DB>>
             <*const [u8] as diesel::deserialize::FromSql<diesel::sql_types::Binary, diesel::sqlite::Sqlite>>
             <*const str as diesel::deserialize::FromSql<diesel::sql_types::Date, diesel::sqlite::Sqlite>>
             <*const str as diesel::deserialize::FromSql<diesel::sql_types::Text, DB>>
           and 3 others
   = note: required because of the requirements on the impl of `diesel::deserialize::FromSql<diesel::sql_types::Integer, _>` for `std::string::String`
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Integer, _>` for `std::string::String`
   = note: required because of the requirements on the impl of `diesel::deserialize::FromSqlRow<diesel::sql_types::Integer, _>` for `std::string::String`
   = note: required because of the requirements on the impl of `diesel::query_dsl::LoadQuery<_, std::string::String>` for `diesel::query_builder::SelectStatement<users::table, diesel::query_builder::select_clause::SelectClause<users::columns::id>>`
